### PR TITLE
Add direct data download support for WSC, MultiRC, and ReCoRD

### PR DESCRIPTION
**Pull request by jeswan**
Some tasks in [`nlp`](https://github.com/huggingface/nlp) are in stored in a different format than the original dataset. In these cases, we download the dataset directly and bypass the `nlp` downloader. For SuperGLUE tasks, it is necessary to directly download the datasets for WSC, MultiRC, and ReCORD. Currently, `tokenize_and_cache.py` fails for these tasks. In the future, it may be desirable to support `nlp` examples directly.

**How were these changes validated?**
All SuperGLUE data was downloaded and configs were generated using the following command:
`python jiant/scripts/download_data/runscript.py download --benchmark SUPERGLUE --output_path $WORKING_DIR/data`
The outputs were inspected manually.

### CODE REVIEW

File: jiant/scripts/download_data/runscript.py [hunk 1]
```diff
@@ -6,9 +6,10 @@
 import jiant.scripts.download_data.datasets.xtreme as xtreme_download
 import jiant.scripts.download_data.datasets.files_tasks as files_tasks_download
 from jiant.tasks.constants import GLUE_TASKS, SUPERGLUE_TASKS, XTREME_TASKS, BENCHMARKS
+from jiant.scripts.download_data.constants import SQUAD_TASKS, INCOMPATIBLE_NLP_TASKS
 
-NLP_DOWNLOADER_TASKS = GLUE_TASKS | SUPERGLUE_TASKS
-SUPPORTED_TASKS = NLP_DOWNLOADER_TASKS | XTREME_TASKS | {"squad_v1", "squad_v2"}
+NLP_DOWNLOADER_TASKS = GLUE_TASKS | SUPERGLUE_TASKS - INCOMPATIBLE_NLP_TASKS
```
Comment (zphang) [inline jiant/scripts/download_data/runscript.py:11]: Do we need to refer to these as "incompatible NLP tasks"? We should simply designate that a certain set of tasks are downloaded via NLP, and the others will be downloaded using other methods.

Comment (jeswan) [reply jiant/scripts/download_data/runscript.py:11]: What do you think about this change?
```suggestion
NLP_DOWNLOADER_TASKS = GLUE_TASKS | SUPERGLUE_TASKS - DIRECT_DOWNLOAD_TASKS
```

**Comment by zphang** [comment]
Is the reason for task insensitive file-name just because the files are upper-cased version of the jiant task names? I'd prefer a more explicit solution (especially when other tasks have similar situations but slightly differently named zips), but happy to just approve for now with an explaining comment.

**Comment by jeswan** [comment]
@zphang For tasks like ReCoRD, the data will be unzipped under `./record/ReCoRD/` and needs to be moved to `./record/`.  I'll add a comment but an explicit mapping should not be necessary since we should keep task names consistent.

**Pull request merged**